MekHQ/src/mekhq/campaign/universe/eras/Eras.java

```diff
@@ -72,14 +72,7 @@ private void setEras(final TreeMap<LocalDate, Era> eras) {
     }
 
     public Era getEra(final LocalDate today) {
-        for (final Map.Entry<LocalDate, Era> entry : getEras().entrySet()) {
-            if (today.isBefore(entry.getKey())) {
-                return entry.getValue();
-            }
-        }
-
-        MekHQ.getLogger().error("Failed to determine a valid era for the date " + today);
-        return getEras().values().stream().findFirst().orElse(new Era());
+        return getEras().ceilingEntry(today).getValue();
```

Comment:
I assume this works for dates past ilClan?

Reply:
ilClan doesn't have an end date, so it uses the default of 01-01-9999... which should handle any dates.